src/modules/new-request-form/NewRequestForm.tsx

```diff
@@ -18,12 +32,43 @@ export function NewRequestForm({
     action,
     http_method,
     accept_charset,
+    errors,
     ticket_form_field,
     ticket_forms_instructions,
   } = requestForm;
 
+  // NOTE: This is a temporary handling of the CSRF token
+  const fetchCsrfToken = async () => {
+    const response = await fetch("/hc/api/internal/csrf_token.json");
+    const { current_session } = await response.json();
+    return current_session.csrf_token as string;
+  };
+
+  const handleSubmit: FormEventHandler<HTMLFormElement> = (e) => {
+    e.preventDefault();
+
+    const form = e.target as HTMLFormElement;
+
+    fetchCsrfToken().then((token) => {
+      const hiddenInput = document.createElement("input");
+      hiddenInput.type = "hidden";
+      hiddenInput.name = "authenticity_token";
+      hiddenInput.value = token;
+      form.appendChild(hiddenInput);
+      form.submit();
+      form.removeChild(hiddenInput);
```

Comment:
Why do we need to remove it?
It shouldn't persist when validation fails, right?

Reply:
In theory, we don't need it, because if there are errors when the form is submitted, the page is reloaded.

I am just not sure if it is always the case, or if it is possible that the `submit` fails and we end up adding multiple times the same hidden input. So I thought it may be helpful to have the field removed to be sure that we don't end up in an ambiguous state.

Actually, I found out that with this implementation the user can click multiple times on the `Submit` button and we end up calling the CSRF token API multiple times. Probably we need to disable the submit button while the token is being fetched or make sure to call the API only once as we are doing in HC